core/iwasm/compilation/aot_emit_aot_file.c

```diff
@@ -2409,7 +2423,10 @@ aot_emit_init_data_section(uint8 *buf, uint8 *buf_end, uint32 *p_offset,
         return false;
 
     offset = align_uint(offset, 4);
-    EMIT_U32(comp_data->func_count);
+    if (need_call_wrapped_indirect(obj_data))
+        EMIT_U32(comp_data->func_count * 2);
+    else
+        EMIT_U32(comp_data->func_count);
```

Comment:
Could we not change here since the above get_func_section_size and below emitting text_offset_of_aot_func_internal/func_type_index are calculated with the original func_count? We can just add reading extra text_offset_of_aot_func_internal and func_type_index in the aot loader.